cmake/onnxruntime_unittests.cmake

```diff
@@ -397,9 +397,7 @@ set(ONNXRUNTIME_TEST_LIBS
     ${ONNXRUNTIME_INTEROP_TEST_LIBS}
     ${onnxruntime_libs}
     ${PROVIDERS_CUDA}
-# These providers are shared libraries now, so aren't linked this way anymore:
-    ${PROVIDERS_DNNL}
-    ${PROVIDERS_TENSORRT}
+    # TENSORRT and DNNL are explicitly linked at runtime
```

Comment:
Would be good to add a reason as to why they're explicitly linked at runtime

Reply:
in the interest of time, I merged it. 
not sure what additional comment is needed compared to the previous comment, since "explicitly linked at runtime" indicates they are shared libs.